clippy_lints/src/comparison_chain.rs

```diff
@@ -84,6 +86,15 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ComparisonChain {
                 {
                     return;
                 }
+
+                // Check that the type being compared implements `core::cmp::Ord`
+                let ty = cx.tables.expr_ty(lhs1);
+                let ord_id = get_trait_def_id(cx, &paths::ORD).unwrap();
```

Comment:
```suggestion
                let Some(ord_id) = get_trait_def_id(cx, &paths::ORD);
```

Reply:
There is no if_chain here, that would make this syntax work. Using `map_or` and a default value of `false` instead.